Linker Error Involving RigidBodyTree<double>::transformQDotMappingToVelocityMapping

# The Error

Using OSX 10.11.6, `clang` version Apple LLVM version 7.3.0 (clang-703.0.29), and SHA cf90c22718c3740617c9702686f28389c692f8f4:

```
FAILED: multibody/rigid_body_plant/libdrakeRigidBodyPlant.dylib
: && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -Werror=all -Werror=ignored-qualifiers -Werror=overloaded-virtual -Werror=shadow -Werror=inconsistent-missing-override -Werror=sign-compare -DGTEST_DONT_DEFINE_FAIL=1 -DGTEST_DONT_DEFINE_SUCCEED=1 -DGTEST_DONT_DEFINE_TEST=1 -O2 -g -DNDEBUG  -dynamiclib -Wl,-headerpad_max_install_names -Wl,-undefined -Wl,error -o multibody/rigid_body_plant/libdrakeRigidBodyPlant.dylib -install_name @rpath/libdrakeRigidBodyPlant.dylib multibody/rigid_body_plant/CMakeFiles/drakeRigidBodyPlant.dir/drake_visualizer.cc.o multibody/rigid_body_plant/CMakeFiles/drakeRigidBodyPlant.dir/viewer_draw_translator.cc.o multibody/rigid_body_plant/CMakeFiles/drakeRigidBodyPlant.dir/contact_detail.cc.o multibody/rigid_body_plant/CMakeFiles/drakeRigidBodyPlant.dir/contact_force.cc.o multibody/rigid_body_plant/CMakeFiles/drakeRigidBodyPlant.dir/contact_info.cc.o multibody/rigid_body_plant/CMakeFiles/drakeRigidBodyPlant.dir/contact_resultant_force_calculator.cc.o multibody/rigid_body_plant/CMakeFiles/drakeRigidBodyPlant.dir/contact_results.cc.o multibody/rigid_body_plant/CMakeFiles/drakeRigidBodyPlant.dir/kinematics_results.cc.o multibody/rigid_body_plant/CMakeFiles/drakeRigidBodyPlant.dir/point_contact_detail.cc.o multibody/rigid_body_plant/CMakeFiles/drakeRigidBodyPlant.dir/rigid_body_plant.cc.o  systems/lcm/libdrakeLCMSystem2.dylib systems/framework/libdrakeSystemFramework.dylib multibody/libdrakeRBM.dylib solvers/libdrakeOptimization.dylib common/trajectories/libdrakeTrajectories.dylib lcm/libdrakeLcm.dylib /Users/liang/dev/drake-distro-1/build/install/lib/liblcm.1.3.1.dylib multibody/collision/libdrakeCollision.dylib /Users/liang/dev/drake-distro-1/build/install/lib/libBulletDynamics.dylib /Users/liang/dev/drake-distro-1/build/install/lib/libBulletCollision.dylib /Users/liang/dev/drake-distro-1/build/install/lib/libLinearMath.dylib /Users/liang/dev/drake-distro-1/build/install/lib/libBulletSoftBody.dylib multibody/shapes/libdrakeShapes.dylib multibody/joints/libdrakeJoints.dylib util/libdrakeGeometryUtil.dylib math/libdrakeMath.dylib multibody/libdrakeXMLUtil.dylib thirdParty/zlib/tinyxml2/libtinyxml2.dylib thirdParty/bsd/spruce/src/libspruce.dylib common/libdrakeCommon.dylib /Users/liang/dev/drake-distro-1/build/install/lib/libgflags.2.2.0.dylib -L/Users/liang/dev/drake-distro-1/build/install/lib -L/usr/local/Cellar/gcc/6.1.0/lib/gcc/6/gcc/x86_64-apple-darwin15.5.0/6.1.0 -L/usr/local/Cellar/gcc/6.1.0/lib/gcc/6/gcc/x86_64-apple-darwin15.5.0/6.1.0/../../.. -lipopt -ldl -lcoinmumps -lgfortran -lSystem -lquadmath -lcoinmetis -lcoinlapack -lcoinblas -lgfortran -lSystem -lquadmath -lcoinmetis -lcoinlapack -lcoinblas /Users/liang/dev/drake-distro-1/build/install/lib/libnlopt.0.9.0.dylib -lm -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/systems/lcm -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/systems/framework -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/multibody -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/solvers -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/common/trajectories -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/lcm -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/install/lib -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/multibody/collision -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/multibody/shapes -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/multibody/joints -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/util -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/math -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/thirdParty/zlib/tinyxml2 -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/thirdParty/bsd/spruce/src -Wl,-rpath,/Users/liang/dev/drake-distro-1/build/drake/common && :
Undefined symbols for architecture x86_64:
  "Eigen::Matrix<Eigen::CwiseNullaryOp<Eigen::internal::scalar_identity_op<double>, Eigen::Matrix<double, -1, -1, 0, -1, -1> >::Scalar, -1, -1, 0, -1, -1> RigidBodyTree<double>::transformQDotMappingToVelocityMapping<Eigen::CwiseNullaryOp<Eigen::internal::scalar_identity_op<double>, Eigen::Matrix<double, -1, -1, 0, -1, -1> > >(KinematicsCache<Eigen::CwiseNullaryOp<Eigen::internal::scalar_identity_op<double>, Eigen::Matrix<double, -1, -1, 0, -1, -1> >::Scalar> const&, Eigen::MatrixBase<Eigen::CwiseNullaryOp<Eigen::internal::scalar_identity_op<double>, Eigen::Matrix<double, -1, -1, 0, -1, -1> > > const&)", referenced from:
      drake::systems::RigidBodyPlant<double>::EvalTimeDerivatives(drake::systems::Context<double> const&, drake::systems::ContinuousState<double>*) const in rigid_body_plant.cc.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
So far, I've tried doing a clean build, which did not resolve the problem.

--- a/drake/multibody/rigid_body_plant/rigid_body_plant.cc
+++ b/drake/multibody/rigid_body_plant/rigid_body_plant.cc
@@ -309,10 +309,12 @@ void RigidBodyPlant<T>::EvalTimeDerivatives(
    * w.r.t. q and v. See issue
    * https://github.com/RobotLocomotion/drake/issues/4267.
    */
+  // TODO(amcastro-tri): Remove .eval() below once RigidBodyTree is fully
+  // templatized.
   const auto& vdot_value =
       drake::solvers::GetSolution(vdot);
   xdot << tree_->transformQDotMappingToVelocityMapping(
-      kinsol, MatrixX<T>::Identity(nq, nq)) * v, vdot_value;
+      kinsol, MatrixX<T>::Identity(nq, nq).eval()) * v, vdot_value;
 
   derivatives->SetFromVector(xdot);
 }

--- a/drake/multibody/rigid_body_system1/RigidBodySystem.cpp
+++ b/drake/multibody/rigid_body_system1/RigidBodySystem.cpp
@@ -268,11 +268,13 @@ RigidBodySystem::StateVector<double> RigidBodySystem::dynamics(
 
   StateVector<double> dot(nq + nv);
 
+  // TODO(amcastro-tri): Remove .eval() below once RigidBodyTree is fully
+  // templatized.
   Eigen::VectorXd vdot_value =
       drake::solvers::GetSolution(vdot);
   dot << tree->transformQDotMappingToVelocityMapping(kinsol,
              Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>::Identity(
-                 nq, nq)) *
+                 nq, nq).eval()) *
              v,
       vdot_value;
   return dot;

